### PR TITLE
When renderer

### PR DESCRIPTION
**Pull request by juwara0**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

Closes: https://github.com/ciena-frost/ember-frost-date-picker/issues/16

**Description:** A new "When" renderer that can be configured to allow for selecting a date via a keyword or a date time string:

```
Turn up
(x) Now
( ) specific date

Turn down
(x) Never
( ) specific date
```

**Requirements:** 
```
"When" renderer
"now" is selected by default
if "now" is selected disable date-time inputs
if "when" (specific date) is selected enable date-time inputs
date-time-picker defaults to current date/time

  configuration
    "now" label is configurable
    "now" value is configurable
    "when" label is not configurable
    "when" value is not configurable
    radio-button size is configurable and defaults to "small"
    dateFormat - is configurable and defaults to "YYYY-MM-DD"
    timeFormat - is configurable and defaults to "HH:mm:ss"
    dateTimeFormat - is configurable and defaults to "YYYY-MM-DDTHH:mm:ssZ"

When values are written to the Bunsen model
  The model property this field was mapped to will be set to one of two values:

    The "now" value supplied in the config as a "string"
    The "when" value of the timestamp "string" 
```
# CHANGELOG
* **Updated** date-time renderer test helpers to be more flexible so they can be used in other tests where date-time-picker is employed
* **Updated** date-time renderer tests to use the more flexible test helpers
* **Added** New `when` renderer that allows the selection of a date based on a keyword or a date time string.
* **Updated** demo app to showcase the new `when` renderer
* **Added** documentation for the new `when` renderer
* **Added** tests for the new `when` render

### CODE REVIEW

File: tests/dummy/app/pods/view/renderers/documentation/when.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    dateTimeFormat: 'MM-DD-YYYYTHH:mm:ssZ',
+    name: 'when'
+  }
+}
+```
+
+#### renderer.label
+
+The label for the first radio button
+```json
```
Comment (agonza40) [inline tests/dummy/app/pods/view/renderers/documentation/when.md]: Nit-picky, but these blocks aren't valid JSON.

File: addon/components/inputs/when.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+    this.setProperties({
+      date,
+      firstButtonValue,
+      selectedValue: firstButtonValue,
+      storedDateTimeValue: moment(currentDateTime).format(this.get('dateTimeFormat')),
+      time
+    })
+
+    run.later(() => {
```
Comment (agonza40) [inline addon/components/inputs/when.js]: Why is this necessary? When did the value change?

File: tests/dummy/app/pods/view/renderers/documentation/when.md (hunk 1 above)
Comment (juwara0) [inline tests/dummy/app/pods/view/renderers/documentation/when.md]: I have made these updates.

File: addon/components/inputs/when.js (hunk 2 above)
Comment (cstolli) [inline addon/components/inputs/when.js]: I'm not sure it is necessary either, pretty sure doing a `setProperties` in `init` does not trigger a re-render anyway, but there might be some reason I'm not thinking of to exhaust the runLoop before calling the `onChange` here, hence the `run.later`

**Comment by cstolli** [comment]
approved

**Comment by cstolli** [comment]
👍

File: addon/components/inputs/when.js (hunk 2 above)
Comment (juwara0) [inline addon/components/inputs/when.js]: This is necessary since the default is to have the first radio button selected we also need to write that value to the bunsen model.

Without wrapping the setting of the initial value to the bunsen model, I get an Ember error that `renderValue` is being modified twice. Once in `component:frost-bunsen-cell` and another in this renderer: `component:frost-bunsen-input-when`.

Comment (cstolli) [inline addon/components/inputs/when.js]: yeah makes sense.
